src/main/resources/assets/computercraft/lua/rom/programs/edit.lua

```diff
@@ -107,7 +107,7 @@ local function save( _sPath )
     if file then 
         file.close()
     end
-    return ok, err
+    return ok, err, fileerr
```

Comment:
Perhaps instead of storing `fileerr` in an upvalue and returning, you should have `error(fileerr, 0)`. That way you don't need the separate `err` and `fileerr` values. It would be worth checking how it handles out of space errors though - those will be generated through `file.write` instead.